eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingServiceComposite.java

```diff
@@ -603,11 +602,25 @@ protected Object calculate() {
                 if (unwrapped != null || copyNamespaces != null) {
                     final MessageComposerType msgComposer = SOAPFactory.eINSTANCE
                             .createMessageComposerType();
-                    if (unwrapped != null) {
+                    final MessageComposerType oldMsgComposer = _binding.getMessageComposer();
+                    boolean unwrappedChanged = true;
+                    if (oldMsgComposer == null && unwrapped != null) {
+                    	unwrappedChanged = true;
+                    } else if (oldMsgComposer != null && oldMsgComposer.isUnwrapped() == unwrapped.booleanValue()) {
+                    	unwrappedChanged = false;
+                    }
+                    if (unwrapped != null && unwrappedChanged) {
                         msgComposer.setUnwrapped(unwrapped.booleanValue());
                     }
-                    if (copyNamespaces != null) {
-                        msgComposer.setCopyNamespaces(copyNamespaces.booleanValue());
+                    boolean copyNamespacesChanged = true;
```

Comment:
what do you think to extract a method isCopyNamespacesChanged(oldMsgComposer,  copyNamespaces)?